perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -251,6 +253,24 @@ public Group createGroup(PerunSession sess, Group parentGroup, Group group) thro
 
 		getPerunBl().getAuditer().log(sess, new GroupCreatedAsSubgroup(group, vo, parentGroup));
 
+		// check if new subgroup should be automatically assigned to resources
+		List<AssignedResource> assignedResources = getPerunBl().getResourcesManagerBl().getResourceAssignments(sess, parentGroup, null)
+			.stream()
+			.filter(AssignedResource::isAutoAssignSubgroups)
+			.collect(toList());
+
+		for (AssignedResource assignedResource : assignedResources) {
+			try {
+				Group sourceGroup = assignedResource.getSourceGroupId() == null ? parentGroup : getPerunBl().getGroupsManagerBl().getGroupById(sess, assignedResource.getSourceGroupId());
```

Comment:
I think `getSourceGroupId()` would always be null because you filtered only assignments with `autoAssignSubgroups` set to true - but only direct assignments can have this set to true. So we probably want to to filter assignments with `autoAssignSubgroups == true` or `getSourceGroupId() != null`. It could be more readable to filter this in the for loop, so we don't need the stream.

Reply:
I believe automatic assignments have `autoAssignSubgroups` set to true too.

Reply:
Yes, you are right.